include/pistache/http_header.h

```diff
@@ -628,19 +628,19 @@ namespace Pistache::Http::Header
         std::string ua_;
     };
 
-#define CUSTOM_HEADER(header_name)                                                  \
-    class header_name : public Pistache::Http::Header::Header                       \
+#define PISTACHE_CUSTOM_HEADER(header_class, header_name)                           \
+    class header_class : public Pistache::Http::Header::Header                      \
```

Comment:
Is renaming the class required? Couldn't you change the macro parameter name instead? It feels like it would cause less backwards compatibility issues to me

Reply:
Do you mean renaming the "macro"? In order to support different class name and header name we'll need 2 different parameter since header names with '-' are not valid class names. So unless compatibility is dropped I need a second macro for the new use case.